ratelimit.go

```diff
@@ -81,6 +86,16 @@ func (rl *ClientRateLimiter) Allow(s string) bool {
 	return present
 }
 
+func (rl *ClientRateLimiter) Delta(s string) time.Duration {
+	rl.RLock()
+	if _, present := rl.bag[s]; !present {
+		rl.RUnlock()
+		return time.Duration(time.Hour * 24)
+	}
+	rl.RUnlock()
+	return rl.bag[s].delta()
```

Comment:
this is a concurrent read access to rl.bag[s], which will result in a panic at runtime